lib/ansible/modules/source_control/gitlab_project_variable.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Markus Bergholz (markuman@gmail.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+module: gitlab_project_variable
+short_description: Creates/updates/deletes Gitlab Projects Variables
+description:
+  - When a project variable does not exists, it will be created.
+  - When a project variable does exists, its value will be updated when the values are different.
+  - Variables which are presented in the playbook, but are not presented in the Gitlab project,
+    they stay untouched (purged_vars = False) or will be deleted (purged_vars = True).
+version_added: "2.9"
+author:
+    - "Markus Bergholz (@markuman)"
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+extends_documentation_fragment:
+  - auth_basic
+options:
+  state:
+    description:
+      - create or delete project variable.
+      - Possible values are present and absent.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  api_token:
+    description:
+      - Gitlab access token with api permissions.
+    required: true
+    type: str
+  project:
+    description:
+      - The path and name of the project
+    required: true
+    type: str
+  purge:
+    description:
+      - When set to true, all variables which are not presented in the task will be deleted.
+    default: false
+    required: false
+    type: bool
+  vars:
+    description:
+      - A list of key value pairs
+    default: {}
+    required: false
+    type: dict
+'''
+
+
+EXAMPLES = '''
+- name: Set or update some CI/CD variables
+  gitlab_project_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: markuman/dotfiles
+    purge: False
+    vars:
+      ACCESS_KEY_ID: abc123
+      SECRET_ACCESS_KEY: 321cba
+
+- name: delete one variable
+  gitlab_project_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: markuman/dotfiles
+    state: absent
+    vars:
+      ACCESS_KEY_ID: abc123
+'''
+
+RETURN = '''
+msg:
+  description: Success or failure message
+  returned: always
+  type: str
+  sample: "Success"
+
+result:
+  description: json parsed response from the server
+  returned: always
+  type: dict
+
+error:
+  description: the error message returned by the Gitlab API
```

Comment:
```suggestion
  description: The error message returned by the Gitlab API
```
The same for the next one.